### PR TITLE
[Bug][Vectorized] fix arithmetic calculate get wrong result

### LINKED ISSUE #8225: [Bug][Vectorized] fix arithmetic calculate get wrong result
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

```sql
select mod(abs(k2), 2), mod(k3, 6), mod(k5, -4) from test where mod(abs(k2), 2) = 0 order by k1, k2, k3, k4,k5;
```
There have some problems:
1. arithmetic function logic is diffrent with arithmetic expr, so i make sql parser parse arithmetic function to arithmetic expr.
2. clang decimal arithmetic calculate get wrong result because of overload error.

### What You Expected?

fix it.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes
@HappenLee 
Issue Number: close #8225

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/modulo.cpp [hunk 1]
```diff
@@ -72,6 +74,7 @@ struct PModuloImpl {
         }
     }
 
+	template <typename Result = DecimalV2Value>
```
Comment (HappenLee) [inline be/src/vec/functions/modulo.cpp:77]: format the code

Comment (BiteTheDDDDt) [reply be/src/vec/functions/modulo.cpp:77]: > format the code

This code already formated.

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -4951,24 +4951,44 @@ window_boundary ::=
 arithmetic_expr ::=
   expr:e1 STAR expr:e2
   {: RESULT = new ArithmeticExpr(ArithmeticExpr.Operator.MULTIPLY, e1, e2); :}
+  | KW_MULTIPLY LPAREN expr:e1 COMMA expr:e2 RPAREN
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/cup/sql_parser.cup:4954]: ```functioname(param1, param2)``` is belongs to FunctionCallExpr .
If each functionname has to add grammar, the grammar file will not be able to be put down long ago

File: be/src/exprs/arithmetic_expr.cpp [hunk 3]
```diff
@@ -21,6 +21,10 @@
 
 namespace doris {
 
+std::set<std::string> ArithmeticExpr::_s_valid_fn_names = {
```
Comment (morningman) [inline be/src/exprs/arithmetic_expr.cpp:24]:
```suggestion
std::unordered_set<std::string> ArithmeticExpr::_s_valid_fn_names = {
```

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**